From Fedora git: http://pkgs.fedoraproject.org/gitweb/?p=sane-backends.git;a=blob;f=sane-backends-1.0.22-v4l.patch;hb=4c4b8135b896ef9f00b926ce5f435ae5c9b156d5

From 26c69b228b29f612faf4b0cc85db969ee7cc4ea6 Mon Sep 17 00:00:00 2001
From: Nils Philippsen <nils@redhat.com>
Date: Mon, 14 Mar 2011 13:35:05 +0100
Subject: [PATCH] patch: v4l

Squashed commit of the following:

commit 23381932c76846191b42a48e505b37cd74711265
Author: Julien BLACHE <jb@jblache.org>
Date:   Wed Feb 16 19:37:43 2011 +0100

    Fix v4l build with libv4l 0.8.3+

    (cherry picked from commit c5ca46c2d1be78c651afb843cc834cf2b5b24953)

    Conflicts:

    	ChangeLog

    Signed-off-by: Nils Philippsen <nils@redhat.com>
---
 backend/v4l.c |    3 +--
 backend/v4l.h |    3 ++-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/backend/v4l.c b/backend/v4l.c
index 6510ef0..38595ed 100644
--- a/backend/v4l.c
+++ b/backend/v4l.c
@@ -84,9 +84,8 @@
 #include "../include/sane/sanei_config.h"
 #define V4L_CONFIG_FILE "v4l.conf"
 
-#include "v4l.h"
-
 #include <libv4l1.h>
+#include "v4l.h"
 
 static const SANE_Device **devlist = NULL;
 static int num_devices;
diff --git a/backend/v4l.h b/backend/v4l.h
index 588b96e..6aee586 100644
--- a/backend/v4l.h
+++ b/backend/v4l.h
@@ -29,6 +29,7 @@
 #ifndef v4l_h
 #define v4l_h
 
+#ifndef __LINUX_VIDEODEV_H
 /* Kernel interface */
 /* Only the stuff we need. For more features, more defines are needed */
 
@@ -165,7 +166,7 @@ struct video_channel
 
 
 /* end of kernel interface */
-
+#endif /* !__LINUX_VIDEODEV_H */
 
 #include <../include/sane/sane.h>
 
-- 
1.7.4

